aten/src/TH/generic/THTensorRandom.cpp

```diff
@@ -434,6 +447,7 @@ void THTensor_(getRNGState)(THGenerator *_generator, THTensor *self)
   THArgCheck(THTensor_(isContiguous)(self), 1, "RNG state needs to be contiguous");
   rng_state = (THGenerator *)THTensor_(data)(self);
   THGenerator_copy(rng_state, _generator);
+  rng_state->mutex = NULL;  // mutex should not be part of the generator state
```

Comment:
Oh god, what this function is even doing (not referring to your change, just a general observation). It's UB at best. We should change it to pack all the data manually, without doing this hack with dumping raw memory into a tensor (this will dump padding too!!). BTW `setRNGState` will copy the bytes of `mutex` from here into the `mutex `field of `_generator`, yielding an invalid state (generator without any mutex) and probably leading to a segfault.